File: src/main/java/net/serebryansky/common/service/TripBuilderService.java
package net.serebryansky.common.service;

import net.serebryansky.builder.model.TripBuilderResponse;
import net.serebryansky.nology.model.Trip;
import reactor.core.publisher.Mono;

public interface TripBuilderService {
    Mono<TripBuilderResponse> buildTrip(Trip trip);
}

File: src/main/kotlin/net/serebryansky/common/service/TripBuilderService.kt
package net.serebryansky.common.service

import net.serebryansky.builder.model.TripBuilderResponse
import net.serebryansky.nology.model.Trip
import reactor.core.publisher.Mono

interface TripBuilderService {
    suspend fun buildTrip(trip: Trip): TripBuilderResponse
}
